messages/btc.proto

```diff
@@ -174,12 +177,19 @@ message BTCSignOutputRequest {
   uint64 value = 3;
   bytes payload = 4; // if ours is false. Renamed from `hash`.
   repeated uint32 keypath = 5; // if ours is true
-  // If ours is true. References a script config from BTCSignInitRequest
+  // If ours is true and `output_script_config_index` is absent. References a script config from
+  // BTCSignInitRequest. This allows change output identification and allows us to identify
+  // non-change outputs to the same account, so we can display this info to the user.
   uint32 script_config_index = 6;
   optional uint32 payment_request_index = 7;
   // If provided, `type` and `payload` is ignored. The generated output pkScript is returned in
   // BTCSignNextResponse. `contains_silent_payment_outputs` in the init request must be true.
   SilentPayment silent_payment = 8;
+  // If ours is true. If set, `script_config_index` is ignored. References an output script config
+  // from BTCSignInitRequest. This enables verification that an output belongs to the same keystore,
+  // even if it is from a different account than we spend from, allowing us to display this info to
+  // the user.
+  optional uint32 output_script_config_index = 9;
```

Comment:
Wondering it this should just be `optional BTCScriptConfigWithKeypath output_script_config` instead of a reference into a vector, to simplify the code and API of client libraries.

The `script_configs` vector for inputs/changes is there so 
- we can validate that multisig/singlesig inputs are not mixed and that different accounts are not mixed, and 
- that we can confirm multisig details at the beginning of a transaction 
- and to have a speedup to not repeat validation for each input
 
All that is not necessary for the output script infos, and the validation performance is not so important there as the user is anyway manually confirming the output :thinking:

Reply:
Makes sense to me :+1: